packages/common/src/store/buy-crypto/sagas.ts

```diff
@@ -197,231 +365,245 @@ function* doBuyCryptoViaSol({
       [connection, connection.getMinimumBalanceForRentExemption],
       0
     )
+    // otherAmountThreshold is the max input amount for the given slippage.
+    // Note: ignores any existing SOL as it should be recovered and swapped into USDC
+    const requiredAmount = Number(quoteResponse.otherAmountThreshold) + minRent
+
+    // Get min stripe purchase amount using USDC as quote for $1
+    const minQuote = yield* call([jupiterInstance, jupiterInstance.quoteGet], {
```

Comment:
The rest of this is largely the same as before but not taking into account existing SOL